### PR TITLE
#100 Refactored Timeframe Buttons to TimeButtons Component from dashboard

### LINKED ISSUE #100: Refactor Timeframe Buttons in Dashboard
# Description
Refactor the timeframe buttons (1 Day, 1 Week, 1 Month, 1 Year) in ```Dashboard.jsx``` into a new ```TimeButtons.jsx``` component. This improves code modularity and maintainability by isolating timeframe selection logic.

# Requirements:
- Create ```TimeButtons.jsx``` with timeframe buttons logic.
- Add ```TimeButtons.css``` for styling.
- Update ```Dashboard.jsx``` to use TimeButtons component.

# How SOLID Principles and CLEAN Architecture are being incorporated:
- **Single Responsibility Principle:** Components have distinct responsibilities, enhancing readability and ease of maintenance.
- **Open/Closed Principle:** New timeframes can be added to TimeButtons without altering existing functionality.
- **CLEAN Architecture:** Ensures separation of concerns by isolating timeframe logic.

# Acceptance Criteria:
- [ ] ```TimeButtons.jsx``` created with timeframe buttons.
- [ ] ```TimeButtons.css``` styles buttons.
- [ ] ```Dashboard.jsx``` uses TimeButtons component.
- [ ] Manual testing confirms correct functionality.

### PR DESCRIPTION
**Pull request by tvirat**
https://github.com/Jai0212/Cash-App-Bias-Busters/issues/100
# Description
Refactor the timeframe buttons (1 Day, 1 Week, 1 Month, 1 Year) in ```Dashboard.jsx``` into a new ```TimeButtons.jsx``` component. This improves code modularity and maintainability by isolating timeframe selection logic.

# Changes:
- Created ```TimeButtons.jsx``` with timeframe buttons logic.
- Added ```TimeButtons.css``` for styling.
- Updated ```Dashboard.jsx``` to use TimeButtons component.

# SOLID Principles and CLEAN Architecture:
- **Single Responsibility Principle:** Each component (TimeButtons and Dashboard) has one clear responsibility, improving readability and maintainability.
- **Open/Closed Principle:** The TimeButtons component can be extended with new features (e.g., more timeframes) without modifying existing code.
- **CLEAN Architecture:** Separation of concerns is achieved by isolating timeframe logic in its own component, following the principles of modular design.

# Testing:
- [ ] Manually tested timeframe buttons.
- [ ] Verified correct timeframe application and UI updates.

### CODE REVIEW

File: frontend/src/pages/DashboardPage/TimeButtons/TimeButtons.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// TimeButtons.jsx
```
Comment (Jai0212) [inline frontend/src/pages/DashboardPage/TimeButtons/TimeButtons.jsx:1]: Is this component reusable?

**Review by Jai0212 — COMMENTED** [review]
Looks good

**Review by Navnoorsinghmahal — COMMENTED** [review]
Looks great

File: frontend/src/pages/DashboardPage/TimeButtons/TimeButtons.jsx (hunk 1 above)
Comment (armaganngul) [reply frontend/src/pages/DashboardPage/TimeButtons/TimeButtons.jsx:1]: That's what I was thinking. If so, this is a really nice implementation so we can adjust the CSS as needed

**Review by armaganngul — COMMENTED** [review]
Looks good!

File: frontend/src/pages/DashboardPage/TimeButtons/TimeButtons.jsx (hunk 1 above)
Comment (tvirat) [reply frontend/src/pages/DashboardPage/TimeButtons/TimeButtons.jsx:1]: Yes, it's reusable

**Pull request merged**